pyproject.toml

```diff
@@ -42,6 +42,10 @@ pytest-repeat = "^0.9.3"
 websockets = "^12.0"
 clickhouse-connect = "^0.7.16"
 kafka-python = "^2.0.2"
+pyyaml = "^6.0.2"
+types-pyyaml = "^6.0.12.20240917"
+testcontainers = "^4.8.1"
```

Comment:
On CI we already run our tests in a docker container which doesn't have access to host docker, so this probably won't work.

If you need to test `sql_exporter` output you can add it to `build-tools` image

Reply:
Docker in Docker is a thing. Is that impossible to support? I'm probably going to keep this code path for running locally if that can't be supported because this is so much easier than compiling sql_exporter for people.